routing/turns_tts_text.cpp

```diff
@@ -47,10 +49,243 @@ void GetTtsText::ForTestingSetLocaleWithJson(std::string const & jsonBuffer, std
   m_getCurLang = platform::ForTestingGetTextByIdFactory(jsonBuffer, locale);
 }
 
+/** Returns -1 for not found, positive for which item in the haystack was found
+ * Finds the needle in an array haystack, and then returns the index of the item found
+ * @tparam N - an arbitrary array size
+ * @param haystack - array of string_views
+ * @param needle - a string to find in the array
+ * @return long - the index of the needle in the haystack
+ */
+template <std::size_t N>
+long FindInStrArray(const std::array<std::string_view, N>& haystack, std::string& needle)
+{
+  auto begin = haystack.begin();
+  for(auto i = begin; i != haystack.end(); ++i) {
+    const std::string_view item = *i;
+    std::string::size_type idx = needle.find(item);
+    if (idx != std::string::npos) {
+      long item_i = std::distance( begin, i );
+      return item_i;
+    }
+  }
+  return -1;
+}
+
+/**
+ * @brief Modifies a string's last character to harmonize its vowel with a -ra/-re suffix
+ * @param std::string & myString - string to harmonize and modify
+ */
+void HungarianBaseWordTransform(std::string& myString) {
+  constexpr std::array<std::string_view, 4> baseStr = {"e", "a", "ö", "ü"};
+  constexpr std::array<std::string_view, 4> harmonyStr = {"é", "á", "ő", "ű"};
+
+  for (size_t i = 0; i < std::size(baseStr); ++i)
+  {
+    if (strings::EndsWith(myString, baseStr[i]))
+      myString.replace(myString.size() - baseStr[i].size(), baseStr[i].size(), harmonyStr[i]);
+  }
+}
+
+/**
+ * @brief Decides if an uppercase/numeric string has a "front" or "back" ending.
```

Comment:
Let's create a separate turns_tts_hungarian.cpp file with all hungarian-related logic, it will make adding other languages easier and general code cleaner.